# HG changeset patch
# Parent 9e9e3508ccb949308183e57bcbc2c1b80a7525c8

Show auto connect setting in STATS connect.

diff -r 9e9e3508ccb9 ircd/s_err.c
--- a/ircd/s_err.c
+++ b/ircd/s_err.c
@@ -458,7 +458,7 @@
 /* 212 */
   { RPL_STATSCOMMANDS, "%s %u %u", "212" },
 /* 213 */
-  { RPL_STATSCLINE, "C %s * %d %d %s %s", "213" },
+  { RPL_STATSCLINE, "C %s %c * %d %d %s %s", "213" },
 /* 214 */
   { 0 },
 /* 215 */
diff -r 9e9e3508ccb9 ircd/s_stats.c
--- a/ircd/s_stats.c
+++ b/ircd/s_stats.c
@@ -107,7 +107,9 @@
       if (tmp->status & CONF_UWORLD)
 	send_reply(sptr, RPL_STATSULINE, host);
       else if (tmp->status & CONF_SERVER)
-	send_reply(sptr, RPL_STATSCLINE, name, port, maximum, hub_limit, get_conf_class(tmp));
+	send_reply(sptr, RPL_STATSCLINE, name,
+	           (tmp->flags & CONF_AUTOCONNECT) ? '+' : '-',
+	           port, maximum, hub_limit, get_conf_class(tmp));
       else if (tmp->status & CONF_CLIENT)
         send_reply(sptr, RPL_STATSILINE,
                    (tmp->username ? tmp->username : ""), (tmp->username ? "@" : ""),
